client/post-editor/editor-post-formats/accordion.jsx

```diff
@@ -1,81 +1,90 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	pick = require( 'lodash/pick' ),
-	find = require( 'lodash/find' ),
-	classNames = require( 'classnames' ),
-	isEmpty = require( 'lodash/isEmpty' );
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { has } from 'lodash';
+import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
-var Accordion = require( 'components/accordion' ),
-	Gridicon = require( 'components/gridicon' ),
-	siteUtils = require( 'lib/site/utils' ),
-	PostFormats = require( './' );
-
-module.exports = React.createClass( {
-	displayName: 'EditorPostFormatsAccordion',
+import Accordion from 'components/accordion';
+import Gridicon from 'components/gridicon';
+import QueryPostFormats from 'components/data/query-post-formats';
+import siteUtils from 'lib/site/utils';
+import PostFormats from './';
+import { getSelectedSiteId, getSelectedSite } from 'state/ui/selectors';
+import { getPostFormats } from 'state/post-formats/selectors';
 
+const EditorPostFormatsAccordion = React.createClass( {
 	propTypes: {
-		site: React.PropTypes.object,
-		post: React.PropTypes.object,
-		postFormats: React.PropTypes.arrayOf( React.PropTypes.shape( {
-			slug: React.PropTypes.string,
-			label: React.PropTypes.string
-		} ) )
+		siteId: PropTypes.number,
+		site: PropTypes.object,
+		post: PropTypes.object,
+		postFormats: PropTypes.object
 	},
 
-	getFormatValue: function() {
-		if ( ! this.props.post ) {
+	getFormatValue() {
+		const { post, site } = this.props;
+		if ( ! post ) {
 			return;
 		}
 
-		if ( this.props.post.format ) {
-			return this.props.post.format;
+		if ( post.format ) {
+			return post.format;
 		}
 
-		return siteUtils.getDefaultPostFormat( this.props.site );
+		return siteUtils.getDefaultPostFormat( site );
 	},
 
-	getSubtitle: function() {
-		var postFormat;
+	getSubtitle() {
+		const formatValue = this.getFormatValue();
+		const { post, postFormats } = this.props;
 
-		if ( ! this.props.post || ! this.props.postFormats ) {
-			return this.translate( 'Loading…' );
+		if ( ! post || ! postFormats ) {
+			return null;
 		}
 
-		postFormat = find( this.props.postFormats, {
-			slug: this.getFormatValue() || 'standard'
-		} );
-
-		if ( postFormat ) {
-			return postFormat.label;
+		if ( has( postFormats, formatValue ) ) {
+			return postFormats[ formatValue ];
 		}
 
-		return this.translate( 'Standard', { context: 'Post format' } );
+		return this.translate( 'Standard', {
+			context: 'Post format'
+		} );
 	},
 
-	render: function() {
-		var classes = classNames( 'editor-post-formats__accordion', this.props.className, {
-			'is-loading': ! this.props.post || ! this.props.postFormats
+	render() {
+		const { className, post, postFormats } = this.props;
+		const classes = classNames( 'editor-post-formats__accordion', className, {
+			'is-loading': ! post || ! postFormats
 		} );
 
-		if ( isEmpty( this.props.postFormats ) ) {
```

Comment:
So I found a subtle regression by this change. It appears that previously we were using this to account for themes which define theme support for `post-formats` but don't define any custom post formats. I notice for the Confit theme, for example, it has `post-formats` support but the `/sites/%s/post-formats` endpoint returns an empty object. This leads to a strange user experience where we show the accordion but there's only a single option:

![image](https://cloud.githubusercontent.com/assets/1779930/19313063/29333218-9063-11e6-8913-68fe12f64465.png)

In master, we'd have avoided the accordion being rendered if there was only one of these options.

So then the question becomes: Do we show the accordion by default, and only hide it after we know that either the theme doesn't support it or there are no custom post formats? Or do we hide it by default and show it only when we know there are custom post formats?

It probably depends on what's more common across themes, since it's more-or-less equally jarring for a control to appear as it is to disappear. Judgment call on your part.

But we should preserve the original behavior, which is to hide it if there are no post formats returned from the endpoint.